addons/mail/static/src/core/common/message.js

```diff
@@ -343,6 +347,16 @@ export class Message extends Component {
         });
     }
 
+    onClickDownloadFiles() {
+        download({
+            data: {
+                file_ids: this.props.message.attachments.map((rec) => rec.id),
+                zip_name: `attachments_${DateTime.local().toFormat("HHmmddMMyyyy")}.zip`,
+            },
+            url: "mail/attachment/zip",
+        });
+    }
```

Comment:
Better move this logic to code of message action `download_files`, so the action is less dependent of the component `Message`.

It still dependents a bit with `component.message`, but at least it's quite minimal... And this is a `message_action`, so relying on just `component.message` is quite decent :)